clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -196,6 +198,16 @@ public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
         return new NetworkClientDelegate.PollResult(timeUntilNextPoll, requests);
     }
 
+    private void maybeFailPendingRequestsOnCoordinatorFatalError() {
```

Comment:
What about moving the entire `maybeFailPendingRequestsOnCoordinatorFatalError()` method into the inner `PendingRequests` class directly? That way the outer class doesn't have to worry about the details of how to fail them, only that it has to call `pendingRequests.maybeFailOnCoordinatorFatalError()`?